traffic_ops/traffic_ops_golang/routes.go

```diff
@@ -21,20 +21,28 @@ package main
 
 import (
 	"crypto/tls"
+	"io"
+	"log"
 	"net"
 	"net/http"
 	"net/http/httputil"
 	"time"
-	"log"
+
 	tclog "github.com/apache/incubator-trafficcontrol/lib/go-log"
-	"io"
 )
 
 var Authenticated = true
 var NoAuth = false
 
+func handlerToFunc(handler http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		handler.ServeHTTP(w, r)
+	}
+}
+
 // Routes returns the routes, and a catchall route for when no route matches.
 func Routes(d ServerData) ([]Route, http.Handler, error) {
+	proxyHandlerFunc := handlerToFunc(rootHandler(d))
```

Comment:
rootHandler should be assigned to its own variable and reused below.
Currently this creates two separate proxy connection handler instance doubling connections made to Mojolicious (well if half the traffic hit api/1.1/servers/status...)

Reply:
ahh...  yes -- you're right